cdap-app-templates/cdap-etl/cdap-etl-realtime/src/test/java/co/cask/cdap/template/etl/realtime/RealtimeElasticsearchSinkTest.java

```diff
@@ -0,0 +1,165 @@
+import co.cask.cdap.template.etl.realtime.sink.RealtimeTableSink;
+import co.cask.cdap.template.etl.realtime.sink.StreamSink;
+import co.cask.cdap.template.etl.realtime.source.DataGeneratorSource;
+import co.cask.cdap.template.etl.realtime.source.JmsSource;
+import co.cask.cdap.template.etl.realtime.source.KafkaSource;
+import co.cask.cdap.template.etl.realtime.source.SqsSource;
+import co.cask.cdap.template.etl.realtime.source.TwitterSource;
+import co.cask.cdap.template.etl.transform.ProjectionTransform;
+import co.cask.cdap.template.etl.transform.ScriptFilterTransform;
+import co.cask.cdap.template.etl.transform.StructuredRecordToGenericRecordTransform;
+import co.cask.cdap.test.AdapterManager;
+import co.cask.cdap.test.SlowTests;
+import co.cask.cdap.test.TestBase;
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.Gson;
+import org.elasticsearch.action.admin.indices.delete.DeleteIndexRequest;
+import org.elasticsearch.action.admin.indices.delete.DeleteIndexResponse;
+import org.elasticsearch.action.delete.DeleteResponse;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.settings.ImmutableSettings;
+import org.elasticsearch.node.Node;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static org.elasticsearch.index.query.QueryBuilders.matchQuery;
+import static org.elasticsearch.node.NodeBuilder.nodeBuilder;
+
+/**
+ * Test for {@link RealtimeElasticsearchSink}
+ */
+public class RealtimeElasticsearchSinkTest extends TestBase {
+  private static final Id.Namespace NAMESPACE = Constants.DEFAULT_NAMESPACE_ID;
+  private static final Id.ApplicationTemplate TEMPLATE_ID = Id.ApplicationTemplate.from("ETLRealtime");
+  private static final Gson GSON = new Gson();
+
+  private Client client;
+  private Node node;
+
+  @ClassRule
+  public static TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Before
+  public void beforeTest() throws Exception {
+    ImmutableSettings.Builder elasticsearchSettings = ImmutableSettings.settingsBuilder()
+      .put("path.data", tmpFolder.newFolder("data"))
+      .put("cluster.name", "testcluster");
+    node = nodeBuilder().settings(elasticsearchSettings.build()).client(false).node();
+    client = node.client();
+  }
+
+  @After
+  public void afterTest() {
+    node.close();
+  }
+
+  @BeforeClass
+  public static void setupTests() throws IOException {
+    // todo: should only deploy test source and cube sink
+    addTemplatePlugins(TEMPLATE_ID, "realtime-sources-1.0.0.jar",
```

Comment:
I believe Ashley has opened a JIRA to address this : https://issues.cask.co/browse/CDAP-3322